data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -53,30 +47,26 @@ private Utils() {
     }
 
     static String post(CloseableHttpClient httpClient, String urlStr, String payload, InputStream stream, long timeoutMs, Map<String, String> headers,
-            boolean leaveOpen)
+                       boolean leaveOpen)
         throws DataServiceException, DataClientException {
         URI url = parseUriFromUrlString(urlStr);
 
         try (InputStream ignored = (stream != null && !leaveOpen) ? stream : null) {
             HttpPost request = setupHttpPostRequest(url, payload, stream, headers);
-            int requestTimeout = timeoutMs > Integer.MAX_VALUE ? Integer.MAX_VALUE : Math.toIntExact(timeoutMs);
-            RequestConfig requestConfig = RequestConfig.custom().setSocketTimeout(requestTimeout).build();
+            RequestConfig requestConfig = RequestConfig.custom().setResponseTimeout(timeoutMs, TimeUnit.MILLISECONDS).build();
             request.setConfig(requestConfig);
 
             // Execute and get the response
-            HttpResponse response = httpClient.execute(request);
+            CloseableHttpResponse response = httpClient.execute(request);
 
-            HttpEntity entity = response.getEntity();
-            if (entity != null) {
-                StatusLine statusLine = response.getStatusLine();
-                String responseContent = EntityUtils.toString(entity);
-                switch (statusLine.getStatusCode()) {
+            if (response != null) {
+                switch (response.getCode()) {
                     case HttpStatus.SC_OK:
-                        return responseContent;
+                        return response.toString();
                     case HttpStatus.SC_TOO_MANY_REQUESTS:
                         throw new ThrottleException(urlStr);
                     default:
-                        throw createExceptionFromResponse(urlStr, response, null, responseContent);
+                        throw createExceptionFromResponse(urlStr, response, null, response.toString());
                 }
             }
```

Comment:
```
 try (CloseableHttpResponse response = httpClient.execute(request)) {
                if (response != null) {
                    String responseContent = EntityUtils.toString(response.getEntity());

                    switch (response.getCode()) {
                        case HttpStatus.SC_OK:
                            return responseContent;
                        case HttpStatus.SC_TOO_MANY_REQUESTS:
                            throw new ThrottleException(urlStr);
                        default:
                            throw createExceptionFromResponse(urlStr, response, null, responseContent);
                    }
                }
            }
```


You need to use Entity to get the actual content of the response, response.toString() just gives the status line.

Fix this and the tests will pass.